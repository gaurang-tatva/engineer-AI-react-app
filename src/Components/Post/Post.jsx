import React from "react";
import { Button, Typography, Card, Box, makeStyles } from "@material-ui/core";
import PropTypes from "prop-types";

const useStyles = makeStyles((theme) => ({
  root: {
    padding: theme.spacing(1, 1),
  },
  descriptionContainer: {
    display: "flex",
    alignItems: "center",
    justifyContent: "space-between",
  },
  buttonContainer: {
    textAlign: "Center",
  },
  elipsis: {
    whiteSpace: "nowrap",
    overflow: "hidden",
    textOverflow: "ellipsis",
  },
}));

/**
 * Post component
 */

function Post({ postData, handleViewMore }) {
  const classes = useStyles();
  
  return (
    <Card className={classes.root}>
      <Box className={classes.titleContainer}>
        <Typography variant="h6" gutterBottom className={classes.elipsis}>
          {postData.title}
        </Typography>
      </Box>
      <Box className={classes.descriptionContainer}>
        <Box className={classes.autherContainer}>
          <Typography variant="subtitle2" gutterBottom>
            Author: <strong>{postData.author} </strong>
          </Typography>
        </Box>
        <Box className={classes.dateContainer}>
          <Typography variant="subtitle2" gutterBottom>
            {postData.created_at}
          </Typography>
        </Box>
      </Box>
      <Box className={classes.buttonContainer}>
        <Button
          variant="contained"
          color="primary"
          onClick={() => handleViewMore(postData)}
        >
          View More
        </Button>
      </Box>
    </Card>
  );
}

Post.propTypes = {
  PostData: PropTypes.object,
  handleViewMore: PropTypes.func,
};

export default Post;
