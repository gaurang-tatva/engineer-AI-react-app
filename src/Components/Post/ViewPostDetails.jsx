import React from "react";
import DialogTitle from "./../Dialog/DialogTitle";
import { withStyles } from "@material-ui/core/styles";
import { makeStyles } from "@material-ui/core/styles";
import MuiDialogContent from "@material-ui/core/DialogContent";
import {
  Table,
  TableBody,
  TableCell,
  TableContainer,
  TableRow,
  Paper,
  Dialog,
} from "@material-ui/core";

const useStyles = makeStyles((theme) => ({
  media: {
    height: 100,
    width: 100,
    borderRadius: theme.spacing(2),
    margin: "0 auto",
  },
}));

const DialogContent = withStyles((theme) => ({
  root: {
    padding: theme.spacing(2),
  },
}))(MuiDialogContent);

/**
 * View order details Modal
 */

export default function ViewOrderDetailModal({
  viewMoreModalData,
  closePostDetailModal,
}) {
  const classes = useStyles();
  let postData = viewMoreModalData.postData;

  return (
    <Dialog
      onClose={closePostDetailModal}
      aria-labelledby="customized-dialog-title"
      open={viewMoreModalData.open}
    >
      <DialogTitle id="customized-dialog-title" onClose={closePostDetailModal}>
        View Details
      </DialogTitle>
      {postData && (
        <DialogContent dividers>
          <TableContainer component={Paper}>
            <Table className={classes.table} aria-label="simple table">
              <TableBody>
                <TableRow>
                  <TableCell>Title</TableCell>
                  <TableCell align="left">{postData.title}</TableCell>
                </TableRow>
                <TableRow>
                  <TableCell>Author</TableCell>
                  <TableCell align="left">{postData.author}</TableCell>
                </TableRow>
                <TableRow>
                  <TableCell>Created</TableCell>
                  <TableCell align="left">{postData.created_at}</TableCell>
                </TableRow>
              </TableBody>
            </Table>
          </TableContainer>
        </DialogContent>
      )}
    </Dialog>
  );
}
