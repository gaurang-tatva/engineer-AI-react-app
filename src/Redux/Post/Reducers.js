import { ACTION_TYPE } from "./Actions";

const initialState = {
  postList: [],
};

/**
 * Post Reducer
 */

const PostReducer = (state = initialState, action) => {
  switch (action.type) {
    case ACTION_TYPE.FETCH_DATA:
      return {
        ...state,
        postList: action.postList,
      };
    default:
      return state;
  }
};

export default PostReducer;
