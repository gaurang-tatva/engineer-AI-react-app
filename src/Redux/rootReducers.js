import Post from "./Post/Reducers";
import { combineReducers } from "redux";

const rootReducer = combineReducers({
  Post,
});

export default rootReducer;
