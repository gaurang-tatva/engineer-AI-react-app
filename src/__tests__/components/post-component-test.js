import React from "react";
import { Provider } from "react-redux";
import Post from "../../Components/Post/Post";
import renderer from "react-test-renderer";
import { store } from "../../Redux/Store";
import postData from "./../../fixture/data.json";

describe("Post Component", () => {
  it("should render without throwing an error", async () => {
    const rendered = renderer.create(
      <Provider dispatch={jest.fn} store={store}>
        <Post postData={postData} handleViewMore={jest.fn} />
      </Provider>
    );
    expect(rendered.toJSON()).toMatchSnapshot();
  });
});
