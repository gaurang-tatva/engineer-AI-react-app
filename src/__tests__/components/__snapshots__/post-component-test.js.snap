// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post Component should render without throwing an error 1`] = `
<div
  className="MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    className="MuiBox-root MuiBox-root-35"
  >
    <h6
      className="MuiTypography-root makeStyles-elipsis-4 MuiTypography-h6 MuiTypography-gutterBottom"
    >
      Polio Vaccine's Efficacy Results Published Today (4/12) in 1955
    </h6>
  </div>
  <div
    className="MuiBox-root MuiBox-root-66 makeStyles-descriptionContainer-2"
  >
    <div
      className="MuiBox-root MuiBox-root-67"
    >
      <h6
        className="MuiTypography-root MuiTypography-subtitle2 MuiTypography-gutterBottom"
      >
        Author: 
        <strong>
          dpflan
           
        </strong>
      </h6>
    </div>
    <div
      className="MuiBox-root MuiBox-root-68"
    >
      <h6
        className="MuiTypography-root MuiTypography-subtitle2 MuiTypography-gutterBottom"
      >
        2020-04-12T16:40:16.000Z
      </h6>
    </div>
  </div>
  <div
    className="MuiBox-root MuiBox-root-69 makeStyles-buttonContainer-3"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        View More
      </span>
    </button>
  </div>
</div>
`;
